### PR TITLE
feat: Added react-router v7 support

### LINKED ISSUE #782: Support for React Router 7
## Context

What's your version of `nuqs`?

```
2.2.2
```

What framework are you using?

- ✅ React Router

Which version of your framework are you using?

```
7.0.1
```

## Description

React router v7 was now released. See: https://reactrouter.com/ 

It seems `react-router-dom` is no longer a thing, everything now lives inside `react-router`. With that said, the existing adapter is no longer compatible with react router 7.

## Reproduction

1. Install react router 7
2. Attempt to use nuqs' react router adapter

### PR DESCRIPTION
**Pull request by teukuamru**
Added support for react-router v7.

Closes #782.

### CODE REVIEW

File: packages/nuqs/adapters/react-router/v6.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// This file is needed for projects that have `moduleResolution` set to `node`
+// in their tsconfig.json to be able to `import {} from 'nuqs/adapters/react-router'`.
```
Comment (franky47) [inline packages/nuqs/adapters/react-router/v6.d.ts:2]: **suggestoion**: could you update the comment path here too please? So it's easier to find & replace all references.

**Review by franky47 — CHANGES_REQUESTED** [review]
LGTM, thanks!

Maybe we could add a comment in the `adapters/react-router` adapter mentioning it's for v6.

Also an update of the docs would be welcome, to point folks to the right adapter (using Callouts in the Remix section).

**Comment by rickyzhangca** [comment]
can't wait!

**Comment by fobos531** [comment]
@franky47 Is there anything blocking this?

@teukuamru I see that the lint action failed - seems pretty easy to fix, the `peerDependencies` just need to be sorted

**Comment by franky47** [comment]
I need to find some free time to try out RRv7, I had an unpushed branch with Remix improvements that could also be factored in this.

**Comment by teukuamru** [comment]
> I need to find some free time to try out RRv7, I had an unpushed branch with Remix improvements that could also be factored in this.

so, should I fix the lint or just wait for you to try out RRv7?

**Comment by franky47** [comment]
I'll do it, I'm also going to add e2e tests for Remix, RRv6 & RRv7.

**Review by franky47 — APPROVED** [review]

**Comment by franky47** [comment]
Merged but I want to add some more tests in a separate PR before shipping it to GA. In the mean time, y'all can use the preview deployment (feedback welcome!):

```
pnpm add https://pkg.pr.new/nuqs@785
```

And import the adapter as such:
```
// app/root.tsx
import { NuqsAdapter } from 'nuqs/adapters/react-router/v7'

// ...

export default function App() {
  return (
    <NuqsAdapter>
      <Outlet />
    </NuqsAdapter>
  )
}
```

**Comment by franky47** [comment]
@faradaytrs Try that URL: `https://pkg.pr.new/nuqs@d000ca8db65a027e11960c4b99736017c2a5ddaa`

I'll check with the pkg.pr.new folks what went wrong with the PR-numbered release, thanks for the heads-up.

**Comment by faradaytrs** [comment]
> @faradaytrs Try that URL: `https://pkg.pr.new/nuqs@d000ca8db65a027e11960c4b99736017c2a5ddaa`
> 
> I'll check with the pkg.pr.new folks what went wrong with the PR-numbered release, thanks for the heads-up.

i upgraded and it works fine, no problems detected

**Comment by ZipBrandon** [comment]
I'm using 2.3.0-beta.2 on a new RR7 app. I incorporated the adapter with `import { NuqsAdapter } from 'nuqs/adapters/react-router/v7'`, but when I use `useQueryState()` I get `Error: useNavigate() may be used only in the context of a <Router> component.`.  I have the Adapter surrounding the outlet in the App.

```tsx
export default function App() {
	return (
		<NuqsAdapter>
			<Outlet />
		</NuqsAdapter>
	);
}```

**Comment by franky47** [comment]
@ZipBrandon thanks for the feedback and testing the beta!

Could you open a dedicated issue with a repro link so I can take a look please?

**Pull request merged**